main_test.go

```diff
@@ -506,6 +506,70 @@ func TestGenerate(t *testing.T) {
       "zone": "uscentral-5"
     }
   }
+}`,
+		},
+		{
+			desc: "happy case with federation support, c2p authority, and xdstp name flag included",
+			input: configInput{
+				xdsServerUri:               "example.com:443",
+				gcpProjectNumber:           123456789012345,
+				vpcNetworkName:             "thedefault",
+				ip:                         "10.9.8.7",
+				zone:                       "uscentral-5",
+				includeV3Features:          true,
+				includeFederationSupport:   true,
+				includeDirectPathAuthority: true,
+				ipv6Capable:                true,
+				includeXDSTPNameInLDS:      true,
+			},
+			wantOutput: `{
+  "xds_servers": [
+    {
+      "server_uri": "example.com:443",
+      "channel_creds": [
+        {
+          "type": "google_default"
+        }
+      ],
+      "server_features": [
+        "xds_v3"
+      ]
+    }
+  ],
+  "authorities": {
+    "": {},
+    "traffic-director-c2p.xds.googleapis.com": {
+      "xds_servers": [
+        {
+          "server_uri": "dns:///directpath-pa.googleapis.com",
+          "channel_creds": [
+            {
+              "type": "google_default"
+            }
+          ],
+          "server_features": [
+            "xds_v3"
+          ]
+        }
+      ]
+    },
+    "trafficdirector.googleapis.com:443": {
+      "client_listener_resource_name_template": "xdstp://traffic-director-global.xds.googleapis.com/envoy.config.listener.v3.Listener/%s"
```

Comment:
Oh boy. This is going to cause confusion. It seems trafficdirector.googleapis.com is wrong, and it should be traffic-director-global.xds.googleapis.com instead. When we do a lookup for which server to request for a CDS resource, it seems we wouldn't find a control plane.

Reply:
I thought I got this verified by the TD team. I'll fix that as part of this PR itself.

Reply:
It is more of a grpc problem than a TD problem. It is completely within the scope of grpc.